addons/mail/models/ir_model.py

```diff
@@ -100,3 +102,29 @@ def _instanciate(self, model_data):
             parents = [parents] if isinstance(parents, str) else parents
             model_class._inherit = parents + ['mail.thread.blacklist']
         return model_class
+
+    def _get_model_definitions(self, model_names_to_fetch):
+        response = {}
+        for model_name in model_names_to_fetch:
+            model = self.env[model_name]
+            all_model_fields = model.fields_get(
```

Comment:
guideline: I like to name indexed dict with the format `stuff_by_index` (`field_by_field_name` in this case, something like that)

the "all_model" in this case is probably overkill as it is in a model loop and there is no other conflicting variable